spacy/matcher.pyx

```diff
@@ -625,8 +628,15 @@ cdef class PhraseMatcher:
             `doc[start:end]`. The `label_id` and `key` are both integers.
         """
         matches = []
-        for _, start, end in self.matcher(doc):
-            ent_id = self.accept_match(doc, start, end)
+        match_doc = doc
+        if self.attr != ORTH:
+            # If we're not matching on the ORTH, match_doc will be a Doc whose
+            # token.orth values are the attribute values we're matching on,
+            # e.g. Doc(nlp.vocab, words=[token.pos_ for token in doc])
+            words = [self.get_lex_value(doc, i) for i in range(len(doc))]
+            match_doc = Doc(self.vocab, words=words)
```

Comment:
I'd probably rather have an else clause here, instead of assigning `match_doc = doc` above, and reassigning in the conditional.